docs/src/pages/docs/functions/helpers.md

```diff
@@ -979,6 +963,46 @@ setPath([ 'list', 'a' ], 'ohhh, I see.', { list: [ 'string', 'another' ] })
 //=> { list: { 0: 'string', 1: 'another', a: 'ohhh, I see.' } }
 ```
 
+#### setProp
+
+`crocks/helpers/setProp`
+
+```haskell
+setProp :: (String | Integer) -> a -> (Object | Array) -> (Object | Array)
+```
+
+Used to set a given value for a specific key or index of
+an `Object` or `Array`. `setProp` takes either a `String` or `Integer` value
+as its first argument and a value of any type as its second. The third parameter
+is dependent of the type of the first argument. When a `String` is provided, the
+third argument must be an `Object`. Otherwise if the first argument is
+an `Integer`, then the third must be an `Array`.
```

Comment:
Is this behaviour what we want, javascript has an interesting feature where keys are forced to be strings regardless of whether they're a number or not. This prevents a scenario where you've created a dictionary for fast access that could be based on an int id where the input is then given as an int. Could we do it the other direction? if it's an array it'll try to make the first arg a number and use is, same for object/string?

Reply:
Hmm. let me think about that one for a bit.
My knee jerk reaction is to keep it explicit as the user should know what they want. OR do not include `Array` as a valid type and keep it as is, only using Array for `Path` functions

Reply:
Nah. I think we should be good with doing it how you suggest.
will update.